src/Components/Web.JS/src/Rendering/WebRendererInteropMethods.ts

```diff
@@ -1,10 +1,16 @@
 import { DotNet } from '@microsoft/dotnet-js-interop';
 import { EventDescriptor } from './Events/EventDelegator';
-import { enableJSRootComponents, JSComponentInfoByInitializer } from './JSRootComponents';
+import { enableJSRootComponents, JSComponentParametersByIdentifier, JSComponentIdentifiersByInitializer } from './JSRootComponents';
 
 const interopMethodsByRenderer = new Map<number, DotNet.DotNetObject>();
 
-export function attachWebRendererInterop(rendererId: number, interopMethods: DotNet.DotNetObject, hasJSComponents: boolean, jsRootComponents: JSComponentInfoByInitializer) {
+export function attachWebRendererInterop(
+  rendererId: number,
+  interopMethods: DotNet.DotNetObject,
+  hasJSComponents: boolean,
+  parameterInfo: JSComponentParametersByIdentifier,
+  initializerInfo: JSComponentIdentifiersByInitializer,
```

Comment:
Would it be OK to give these two parameters a name that's more explicitly related to the JS root components feature?

The reason is that this `attachWebRendererInterop` API is a kind of general purpose "set up everything" function, not tied to a specific feature, so it won't necessarily be super clear to future developers working on this what `parameterInfo` or `initializerInfo` relate to.

It would be even more ideal if we could coalesce all of the JS root components related info into a single parameter. If we have two different things both keyed by identifier (parameterinfo and initializerinfo), would it make sense to rejig the data structure a bit so there's a single dictionary like `identifier => (parameterinfo, initializerinfo)` instead of two dictionaries? Not a big deal in any way since this is an internal API and we can change it later if we want. If you prefer not to have this PR expand in scope and want to get back to the Angular parts I totally understand if you prefer not to move more things around.

Reply:
Also it should now be possible to remove the `hasJSComponents` arg because the JS code can infer that based on whether `parameterInfo` is empty or not (previously there was no way to know if there were JS root components without initializers, but since you're now passing the params info in all cases, we can know based on that).